src/text_shaper/open_shaper.cpp

```diff
@@ -423,9 +423,6 @@ struct open_shaper::Private // {{{
     optional<font_key> getOrCreateKeyForFont(font_source const& source, font_size fontSize)
     {
         auto const sourceId = identifierOf(source);
-        if (auto i = fontPathAndSizeToKeyMapping.find(FontPathAndSize { sourceId, fontSize });
-            i != fontPathAndSizeToKeyMapping.end())
-            return i->second;
```

Comment:
Myconcern I am having (right now) is, that it's not a get-or-create anymore, but you'll ALWAYS create a new key, right?

Maybe the cache key is wrong then. If a font file can contain more than one face (or weight), but a font_key should represent a specific "font face", then we need to encode the weight into it too.

Why I didn't do that so far is, because (to my knowledge), a font file can only contain one weight (and style). 🤔

Reply:
> Why I didn't do that so far is, because (to my knowledge), a font file can only contain one weight (and style). 🤔

[variable fonts](https://en.wikipedia.org/wiki/Variable_font) are fairly new, and that is usually a reasonable assumption.

Reply:
Variable fonts are fonts that support arbitrary weights though interpolation, but the problem also happens with fonts distributed as TrueType collections (usually `.ttc`), like Iosevka.

Reply:
yes, so basically we need to include calculation of weight in the font_face or just renema this function and create new key always

Reply:
> include calculation of weight in the font_face

Would that work with TrueType Collections too? Typically those contain normal styles and fixed weights, just bundled in one file for file size reasons

Reply:
Yes, we can distinguish between them, at the moment we only use font name and size to check if we already loaded this font, so we 
 need to add font weight into comparison

Reply:
So i updated PR and now we check for caching but taking into account weight of the font